.github/workflows/github_server.yml

```diff
@@ -0,0 +1,77 @@
+name: GitHub server tests
+
+on:
+  push:
+  workflow_dispatch:
+    inputs:
+      tags:
+        required: False
+        description: 'Optional. Use a "tag expression" specify which tagged tests to run (https://cucumber.io/docs/cucumber/api/#tag-expressions)'
+      enable_tmate:
+        type: boolean
+        description: 'Run the build with tmate debugging enabled (https://github.com/marketplace/actions/debugging-with-tmate)'
+        required: false
+        default: false
+      # TODO: alkiln_version:
+
+jobs:
+
+  github-server-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [18.x]
+
+    env:
+      # TODO: Must solve - how would a dev make another user/password if they wanted to?
+      SECRET_VAR1: secret-var1-value
+      SECRET_VAR2: secret-var2-value
+      SECRET_FOR_MISSING_FIELD: secret for missing field
+      
+    steps:
+      # Place the root directory in this branch
+      - uses: actions/checkout@v3
+
+      # Debugging. Maybe add a long sleep after whatever it is we want to check
+      # to give us time to explore.
+      - name: Setup tmate session
+        if: ${{ inputs.enable_tmate }}
+        uses: mxschmitt/action-tmate@v3
+        with:
+          detached: true
+
+      - name: Start the temporary GitHub server
+        id: github_server
+        uses: ./github_server
+        with:
+          CONFIG_CONTENTS: "${{ secrets.CONFIG_CONTENTS }}"
```

Comment:
Would it be useful to add a default config that's the simplest config ever? I think it would only need 2 or 3 props. That way this can be run immediately if there aren't special needs.

Reply:
DA has a default config in the docker inage. If we just don't set the config when starting the DA container, we get https://github.com/jhpyle/docassemble/blob/master/Docker/config/config.yml.dist.

Reply:
Makes sense! Then we just need to make sure to only conditionally create the config file if the config secret exists.

Reply:
This now handles no config and, supposedly, handles a config with a value and passes. It also logs the message I created for a non-existent config: "The developer did not provide a custom docassemble config file as an input into this action. Docassemble will use its default config.".

Not sure how to confirm that we use the right config when we do have a config, but I restored our config and saw the console message I created for an existing config: "The developer did provided a custom docassemble config file as an input into this action."